utils/mermaid.js

```diff
@@ -0,0 +1,202 @@
+const GRAPH_TITLE = new String("graph TB\n");
+const FULL_DEPTH = document.getElementById("depth-select").options[0];
+const COMMENT = new String("%% ");
+const FUNC_ID_PATTERN = /(?<node_depth>\d+)_(?<node_id>\d+)/;
+let startId = new Array();
+let startDepth = new Array();
+let startName = new Array();
+let callFlow = new Array();
+let openNode = new Array();
+let baseRootName = new String();
+let baseRootId = new String("0");
+let baseGraph = new String();
+let baseRootDepth = new Number(FULL_DEPTH);
+let showHideDesc = new String();
+let maxDepth = new Number(1);
+
+const mermaidTxt = document.getElementById("meraid_graph").innerHTML;
+
+var showHideNode = function(nodeId) {
+	const nodeIdResult = nodeId.match(FUNC_ID_PATTERN);
+	if (nodeIdResult == null) {
+		document.getElementById("description").innerHTML = "Invalid Node ID : " + nodeId;
+		return;
+	}
+	const id = nodeIdResult.groups.node_id;
+	const depth = nodeIdResult.groups.node_depth;
+	const open = openNode.includes(id);
+	let start = new Boolean(false);
+	for (let i = 0; i < callFlow.length; i++) {
+		if (start == false && Number(startId[i]) == id) {
+			start = true;
+		}
+		if (start == true) {
+			if (Number(startId[i]) == id || startDepth[i] > depth) {
+				if (open)
+					callFlow[i] = COMMENT + callFlow[i];
+				else
+					callFlow[i] = callFlow[i].replace(COMMENT, "");
+			}
+			else {
+				break;
+			}
+		}
+	}
+	if (open)
+		openNode.pop(id);
+	else
+		openNode.push(id);
+
+	renderGraph("".concat(GRAPH_TITLE, callFlow.join("\n"), "\n", showHideDesc));
+};
+
+var selectSubGraph = function(nodeId) {
+	const nodeIdResult = nodeId.match(FUNC_ID_PATTERN);
+	const id = nodeIdResult.groups.node_id;
+
+	for (let i = 0; i < callFlow.length; i++) {
+		if (Number(startId[i]) == id) {
+			drawBaseGraph(id);
+			break;
+		}
+	}
+};
+
+var initializeBaseGraph = function() {
+	startId.length = 0;
+	startDepth.length = 0;
+	startName.length = 0;
+	openNode.length = 0;
+	callFlow.length = 0;
+	showHideDesc = "";
+	baseGraph = "";
+};
+
+var getNodeStyle = function(id, fillColor, strokeColor) {
+	return "  style " + id + " fill:" + fillColor + " ,stroke:" + strokeColor +
+	       ",stroke-width:px;";
+};
+
+var drawDepthGraph = function(depth) {
+	mermaid.initialize(config);
+	showHideDesc = "";
+	openNode.length = 0;
+	const nodeList = new Array();
+	let id = new String();
+	for (let i = 0; i < callFlow.length; i++) {
+		callFlow[i] = callFlow[i].replace(COMMENT, "");
+		if (Number(startDepth[i]) >= depth)
+			callFlow[i] = (COMMENT + callFlow[i]);
+		if (Number(startDepth[i]) == depth) {
+			id = startDepth[i] + "_" + startId[i];
+			if (nodeList.includes(id) == false) {
+				nodeList.push(id);
+				showHideDesc =
+					showHideDesc.concat(getNodeStyle(id, "#f9f", "#333"), "\n");
```

Comment:
It'd be nice if we have a symbolic name for this constant so that one can customize it easily.

Reply:
Thank you for the information!
I updated it to 1024

And I changed this color value to use a constant definition. It looks cleaner. Thank you.